src/Build/BackEnd/Components/SdkResolution/SdkResolverCachingWrapper.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Collections;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Build.BackEnd.SdkResolution
+{
+
+    internal sealed class SdkResolverCachingWrapper: ISdkResolverService
```

Comment:
Do we want the evaluation context to also be reused for builds?  For example I create the evaluation service, evaluate some projects and then build but the SDK resolver cache won't be re-used because the submission ID will be set.  If so, this wrapper could just use a const for submission ID.

Reply:
Builds based on Project objects do not reevaluate. Instead, they execute their targets based on the prior evaluated state from the Project object. However, msbuild task invocations from p2ps (like `GetTargetPath` ) do get evaluated from scratch. So rather than transferring sdk resolution results from Project evaluations to builds, the bigger problem to solve here is how to pass a pool of evaluated Projects to a submission build, such that the p2ps do not get reevaluated but instead reuse the prior in-memory evaluations.